microproxy/config.py

```diff
@@ -6,52 +6,8 @@
 logger = get_logger(__name__)
 
 
-class Config(object):
-    def __init__(self, config_field_info, file_config, cmd_config):
-        command_type = cmd_config["command_type"]
-        optionInfos = config_field_info[command_type]
-
-        file_config = {k.replace(".", "_"): v for k, v in file_config.items(command_type)}
-        cmd_config = {k: v for k, v in cmd_config.iteritems() if v is not None and k != "config_file"}
-        self.__config = file_config.copy()
-        self.__config.update(cmd_config)
-        self.__config.update(self.appendDefault(self.__config, optionInfos))
-        self.__config.update(self.typeTransform(self.__config, optionInfos))
-
-    def appendDefault(self, config, optionInfo):
-        fields = [field for field in optionInfo if field not in config and not optionInfo[field]["is_require"]]
-        options = {field: optionInfo[field]["default"] for field in fields}
-        return options
-
-    def typeTransform(self, config, optionInfo):
-        new_config = {}
-        for field in config:
-            if field not in optionInfo:
-                continue
-
-            if optionInfo[field]["type"] == "int":
-                new_config[field] = int(config[field])
-
-            elif optionInfo[field]["type"] == "list":
-                values = [value for value in config[field].split(",") if len(value) > 0]
-                if optionInfo[field]["list_type"] == "int":
-                    new_config[field] = map(int, values)
-                elif optionInfo[field]["list_type"] == "string":
-                    new_config[field] = values
-
-            else:
-                new_config[field] = config[field]
-
-        return new_config
-
-    def __getitem__(self, key):
-        try:
-            return self.__config[key]
-        except KeyError:
-            raise
-
-    def __iter__(self):
-        return iter(self.__config)
+class Config(dict):
```

Comment:
The previous Config with only implement __getitem__, __iter__ let me hard to copy and override another instance.
```python
class ReplayHandler(object):
    def __init__(self, config, proxy_server):
        config_dict = dict(config)
        config_dict.update(dict(mode="replay"))
        self.config = Config(config_dict)
        self.proxy_server = proxy_server
```

And actually, I think config is just dictionary, so maybe it's no need to create a class for it.
Or it should provide good function that need create a class for it.